numpy/core/src/umath/ufunc_object.c

```diff
@@ -1730,12 +1764,17 @@ execute_fancy_ufunc_loop(PyUFuncObject *ufunc,
         }
     }
     for (i = nin; i < nop; ++i) {
+        /*
+         * Because we don't write to all elements, the output arrays
+         * must be considered READWRITE by the iterator.
```

Comment:
Updateifcopy copies all elements back, not only those for which wheremask is true, so the old content of the output array must be copied there.